testing-common/src/main/java/io/opentelemetry/instrumentation/test/InMemoryExporter.java

```diff
@@ -5,240 +5,93 @@
 
 package io.opentelemetry.instrumentation.test;
 
-import com.google.common.base.Predicate;
-import com.google.common.base.Predicates;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.toList;
+
 import com.google.common.base.Stopwatch;
 import com.google.common.collect.Iterables;
 import com.google.common.collect.TreeTraverser;
-import com.google.common.primitives.Ints;
-import com.google.common.primitives.Longs;
 import io.opentelemetry.api.trace.SpanId;
-import io.opentelemetry.context.Context;
-import io.opentelemetry.sdk.common.CompletableResultCode;
-import io.opentelemetry.sdk.trace.ReadWriteSpan;
-import io.opentelemetry.sdk.trace.ReadableSpan;
-import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.javaagent.testing.common.AgentTestingExporterAccess;
+import io.opentelemetry.sdk.metrics.data.MetricData;
 import io.opentelemetry.sdk.trace.data.SpanData;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.Comparator;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.CopyOnWriteArrayList;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
-import java.util.concurrent.atomic.AtomicInteger;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-public class InMemoryExporter implements SpanProcessor {
-
-  private static final Logger log = LoggerFactory.getLogger(InMemoryExporter.class);
-
-  private final List<List<SpanData>> traces = new ArrayList<>(); // guarded by tracesLock
-
-  private boolean needsTraceSorting; // guarded by tracesLock
-  private final Set<String> needsSpanSorting = new HashSet<>(); // guarded by tracesLock
-
-  private final Object tracesLock = new Object();
-
-  // not using span startEpochNanos since that is not strictly increasing so can lead to ties
-  private final Map<String, Integer> spanOrders = new ConcurrentHashMap<>();
-  private final AtomicInteger nextSpanOrder = new AtomicInteger();
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
 
-  private volatile boolean forceFlushCalled;
+public class InMemoryExporter {
```

Comment:
Technically it's not an exporter but we can cleanup in followups

Reply:
👍 #1955